src/common.c

```diff
@@ -185,6 +185,12 @@ int ofi_ep_bind_valid(struct fi_provider *prov, struct fid *bfid, uint64_t flags
 			return -FI_EBADFLAGS;
 		}
 		break;
+	case FI_CLASS_SRX_CTX:
+		if (flags & ~(FI_RECV)) {
+			FI_WARN(prov, FI_LOG_EP_CTRL, "invalid SRX ctx flags\n");
+			return -FI_EBADFLAGS;
+		}
+		break;
```

Comment:
The fi_endpoint man page doesn't mention requiring/using flags when binding a shared context to an endpoint. So I believe this check would not be needed. The app should not pass any flags.
@j-xiong can you please confirm?